File: 5.02_-_2nd_lab/app/src/main/java/com/example/a502_2ndlab/SecondActivity.java
package com.example.a502_2ndlab;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

import androidx.appcompat.app.AppCompatActivity;

public class SecondActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);

        EditText editText = findViewById(R.id.editText);
        Button buttonThirdActivity = findViewById(R.id.buttonThirdActivity);

        buttonThirdActivity.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String text = editText.getText().toString();

                Intent intent = new Intent(SecondActivity.this, ThirdActivity.class);
                intent.putExtra("text", text);
                startActivity(intent);
            }
        });
    }
}

File: 2504/app/src/main/java/com/example/a2504/SecondActivity.kt
package com.example.a2504

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText

class SecondActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)

        val editText = findViewById<EditText>(R.id.editText)
        val buttonThirdActivity = findViewById<Button>(R.id.buttonThirdActivity)

        buttonThirdActivity.setOnClickListener {
            val text = editText.text.toString()
            val intent = Intent(this, ThirdActivity::class.java)
            intent.putExtra("text", text)
            startActivity(intent)
        }
    }
}